pkg/servicemesh/federation/server/routing.go

```diff
@@ -0,0 +1,275 @@
+		}
+	}
+	return nil
+}
+
+func (s *meshServer) createOrUpdateAuthorizationPolicy(target *federationmodel.ServiceMessage) error {
+	// AuthorizationPolicy used to restrict inbound requests to known clients accessing exported services.
+	// We use a DENY policy to block any traffic coming in that's not from a known client or destined for an exported service
+	name := fmt.Sprintf("federation-exports-%s", s.mesh.Name)
+	rawAP := s.configStore.Get(collections.IstioSecurityV1Beta1Authorizationpolicies.Resource().GroupVersionKind(), name, s.mesh.Namespace)
+	if rawAP == nil {
+		if s.mesh.Spec.Security == nil || s.mesh.Spec.Security.ClientID == "" {
+			logger.Errorf("no ClientID specified for MeshFederation %s/%s: AuthorizationPolicy for exported services will not be created", s.mesh.Namespace, s.mesh.Name)
+			return nil
+		}
+		ap := &config.Config{
+			Meta: config.Meta{
+				GroupVersionKind: collections.IstioSecurityV1Beta1Authorizationpolicies.Resource().GroupVersionKind(),
+				Name:             name,
+				Namespace:        s.mesh.Namespace,
+			},
+			Spec: &rawsecurity.AuthorizationPolicy{
+				Selector: &rawtype.WorkloadSelector{
+					MatchLabels: map[string]string{
+						"service.istio.io/canonical-name": s.mesh.Spec.Gateways.Ingress.Name,
+					},
+				},
+				Action: rawsecurity.AuthorizationPolicy_DENY,
+				Rules: []*rawsecurity.Rule{
+					{
+						From: []*rawsecurity.Rule_From{
+							{
+								Source: &rawsecurity.Source{
+									NotPrincipals: []string{
+										s.mesh.Spec.Security.ClientID,
+									},
+								},
+							},
+						},
+						To: []*rawsecurity.Rule_To{
+							{
+								Operation: &rawsecurity.Operation{
+									NotHosts: []string{
+										target.Hostname,
+									},
+									Ports: []string{
+										strconv.FormatInt(common.FederationPort, 10),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		if _, err := s.configStore.Create(*ap); err != nil {
+			return err
+		}
+		return nil
+	}
+	apSpec := rawAP.Spec.(*rawsecurity.AuthorizationPolicy)
+	if len(apSpec.Rules) != 1 || len(apSpec.Rules[0].To) != 1 {
+		return fmt.Errorf("invalid AuthorizationPolicy for service export")
+	}
+	for _, host := range apSpec.Rules[0].To[0].Operation.NotHosts {
+		if host == target.Hostname {
+			// no update required
+			return nil
+		}
+	}
+	// add the host to the list of available targets
+	apSpec.Rules[0].To[0].Operation.NotHosts = append(apSpec.Rules[0].To[0].Operation.NotHosts, target.Hostname)
+	if _, err := s.configStore.Update(*rawAP); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *meshServer) gatewayForExport(source serviceKey, target *federationmodel.ServiceMessage) *config.Config {
+	resourceName := createResourceName(s.mesh.Name, source)
+	mode := rawnetworking.ServerTLSSettings_ISTIO_MUTUAL
+	if s.mesh.Spec.Security != nil && s.mesh.Spec.Security.AllowDirectInbound {
+		// XXX: this will not work, as the exported services will have a different domain suffix
```

Comment:
yeah, for this we would need to keep the special cluster naming around (what I did with the export label)